pkg/ipamd/ipamd.go

```diff
@@ -1003,66 +1056,55 @@ func (c *IPAMContext) eniIPPoolReconcile(ipPool []string, attachedENI awsutils.E
 				log.Debugf("Reconcile skipping IP %s on ENI %s because it was recently unassigned from the ENI.", strPrivateIPv4, eni)
 				continue
 			} else {
-				log.Debugf("This IP was recently freed, but is out of cooldown. We need to verify with EC2 control plane.")
-				// Call EC2 to verify IPs on this ENI
-				ec2Addresses, err := c.awsClient.GetIPv4sFromEC2(eni)
-				if err != nil {
-					log.Error("Failed to fetch ENI IP addresses!")
-					continue
-				} else {
+				if needEC2Reconcile {
+					// IMDS data might be stale
+					log.Debugf("This IP was recently freed, but is now out of cooldown. We need to verify with EC2 control plane.")
+					// Only call EC2 once for this ENI
+					if ec2VerifiedAddresses == nil {
+						var err error
+						// Call EC2 to verify IPs on this ENI
+						ec2VerifiedAddresses, err = c.awsClient.GetIPv4sFromEC2(eni)
```

Comment:
Can we move this to line 1044? if `needEC2Reconcile` is set then we can make just one call to ec2 to get all ec2 verified IPs. Please correct me if I am wrong.

Reply:
You are right, we just want to do this call once. That's why we have `ec2VerifiedAddresses` initialized to `nil`. If we have an IP out of cooldown, we call EC2 and update `ec2VerifiedAddresses`. The next time, the variable is already populated.

Reply:
Ahh ok got it :)